include/NovelRT/Maths/GeoVector2.h

```diff
@@ -0,0 +1,197 @@
+// Copyright © Matt Jones and Contributors. Licensed under the MIT Licence (MIT). See LICENCE.md in the repository root for more information.
+
+#ifndef NOVELRT_MATHS_GEOVECTOR2_H
+#define NOVELRT_MATHS_GEOVECTOR2_H
+
+#ifndef NOVELRT_H
+#error Please do not include this directly. Use the centralised header (NovelRT.h) instead!
+#endif
+
+namespace NovelRT::Maths {
+  template<typename T>
+  class GeoVector2 {
+    friend class Graphics::RenderObject;
+    friend class Input::InteractionService;
+    friend class GeoBounds;
+    template<typename U>
+    friend class GeoMatrix4;
+
+  private:
+    GeoVector2(glm::vec<2, T> value) : _value(glm::vec<4, T>(value, 0, 0)) {}
+    GeoVector2(glm::vec<4, T> value) : _value(value) {}
```

Comment:
Now that we have `GeoVector4`, this type should likely explicitly wrap a `glm::vec<2, T>` and we should have a `GeoVector4(GeoVector2 value, T z = 0, T w = 0)` overload (and similar for `GeoVector3`) instead.

It allows space savings, simplifies various bits, and makes the upcast to `vec<4, T>` explicit.